myCryptoWallet-ConsoleApp/src/main/java/com/isa/CoinSearch.java

```diff
@@ -0,0 +1,43 @@
+package com.isa;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+
+
+public class CoinSearch {
+
+    public static Coin[] readCoinsFromJson() {
+        try {
+            Coin[] coins = Data.deserializeCoin();
+            return coins;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static List<Coin> search(Coin[] coins, String searchCriteria) {
+        List<Coin> results = new ArrayList<>();
+        for (Coin coin : coins) {
+            if (coin.getSymbol().contains(searchCriteria)){
+                results.add(coin);
+            }
+        }
+        return results;
+    }
+
+    public static void findYourToken() {
+        Scanner sc = new Scanner(System.in);
+        Coin[] coinList = readCoinsFromJson();
+        System.out.println("Podaj kryterium wyszukiwania (symbol):");
+        String searchCriteria = sc.nextLine();
```

Comment:
Ja bym tu dodał metodę toUpperCase(). Bo jeśli wyszukuję np "xrp" z małych liter to symbol się nie pojawia mimo iż jest na liście. A ponieważ wszystkie symbole są z dużych liter to myślę, że to może ułatwić sprawę.
`String searchCriteria = sc.nextLine().toUpperCase();`